src/train.py

```diff
@@ -0,0 +1,262 @@
+    train_params.add_argument("--env", type=str, default='BreakoutDeterministic-v4', help="Environment to use (must have RGB image state space and discrete action space)")
+    train_params.add_argument("--render", type=bool, default=False, help="Whether or not to display the environment on the screen during training")
+    train_params.add_argument("--random_seed", type=int, default=1234, help="Random seed for reproducability")
+    train_params.add_argument("--frame_width", type=int, default=105, help="Frame width after resize.")
+    train_params.add_argument("--frame_height", type=int, default=80, help="Frame height after resize.")
+    train_params.add_argument("--frames_per_state", type=int, default=4, help="Sequence of frames which constitutes a single state.")
+    
+    # Training parameters
+    train_params.add_argument("--num_steps_train", type=int, default=50000000, help="Number of steps to train for")    
+    train_params.add_argument("--train_frequency", type=int, default=4, help="Perform training step every N game steps.")
+    train_params.add_argument("--max_ep_steps", type=int, default=2000, help="Maximum number of steps per episode")
+    train_params.add_argument("--batch_size", type=int, default=32)
+    train_params.add_argument("--learning_rate", type=float, default=0.00025)
+    train_params.add_argument("--replay_mem_size", type=int, default=1000000, help="Maximum size of replay memory buffer")
+    train_params.add_argument("--initial_replay_mem_size", type=int, default=50000, help="Initial size of replay memory (populated by random actions) before learning can start")
+    train_params.add_argument("--epsilon_start", type=float, default=1.0, help="Exploration rate at the beginning of training.")
+    train_params.add_argument("--epsilon_end", type=float, default=0.1, help="Exploration rate at the end of decay.")
+    train_params.add_argument("--epsilon_step_end", type=int, default=1000000, help="After how many steps to stop decaying the exploration rate.")
+    train_params.add_argument("--discount_rate", type=float, default=0.99, help="Discount rate (gamma) for future rewards.")
+    train_params.add_argument("--update_target_step", type=float, default=10000, help="Copy current network parameters to target network every N steps.")
+    train_params.add_argument("--save_ckpt_step", type=float, default=250000, help="Save checkpoint every N steps")
+    train_params.add_argument("--save_log_step", type=int, default=1000, help="Save logs every N steps")
+    
+    # Files/directories
+    train_params.add_argument("--ckpt_dir", type=str, default='./ckpts', help="Directory for saving/loading checkpoints")
+    train_params.add_argument("--ckpt_file", type=str, default=None, help="Checkpoint file to load and resume training from (if None, train from scratch)")
+    train_params.add_argument("--log_dir", type=str, default='./logs/train', help="Directory for saving logs")
+    
+    return train_params.parse_args()
```

Comment:
we might not need all of these and some of the defaults might not make sense (env, frame width, frame height)